spec/models/post_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'associations' do
+    it 'belongs to author' do
+      post = Post.reflect_on_association(:author)
+      expect(post.macro).to eq(:belongs_to)
+    end
+
+    it 'has many comments' do
+      post = Post.reflect_on_association(:comments)
+      expect(post.macro).to eq(:has_many)
+    end
+
+    it 'has many likes' do
+      post = Post.reflect_on_association(:likes)
+      expect(post.macro).to eq(:has_many)
+    end
+  end
+
+  describe 'validations' do
+    before(:each) do
+      @user = User.create(name: 'User 1', posts_counter: 0)
+    end
+    it 'is valid with valid attributes' do
+      post = Post.new(title: 'Post 1', author: @user, comments_counter: 0, likes_counter: 0)
+      expect(post).to be_valid
+    end
+
+    it 'is not valid without a title' do
+      post = Post.new(title: nil, author: @user, comments_counter: 0, likes_counter: 0)
+      expect(post).to_not be_valid
+    end
+
+    it 'is not valid with a negative comments_counter' do
+      post = Post.new(title: 'Post 1', author: @user, comments_counter: -1, likes_counter: 0)
+      expect(post).to_not be_valid
+    end
+
+    it 'is not valid with a negative likes_counter' do
+      post = Post.new(title: 'Post 1', author: @user, comments_counter: 0, likes_counter: -1)
+      expect(post).to_not be_valid
+    end
+  end
+
+  describe 'recent_comments' do
+    it 'returns the 5 most recent comments' do
+      user = User.create(name: 'User 1')
+      user.posts.create(title: 'Post 1', text: 'Post 1 text')
+      post.comments.create(text: 'Comment 1', author: user) # comment1
```

Comment:
- [ ] Almost all the tests passed, but one test failed. I think you need to define a variable post before using it. I have attached an image for your reference.

![post](https://user-images.githubusercontent.com/53124548/229105176-9a06c529-51a5-42e7-8b2f-3a7c377d6444.png)

Reply:
- Thank you, I must have deleted it while fixing the linter errors.